scripts/lint-mdc.ts

```diff
@@ -496,28 +757,234 @@ function checkOneClassMemberIsInReadme(
     logLinterViolation(
         inputFilePath,
         loc,
-        `Public ${kind} '${name}' is missing from '${readmeFilePathRelative}'.`,
+        [
+          `Missing documentation for ${kind}: '${name}'.`,
+          `Most publicly exported symbols must be documented in '${readmeFilePathRelative}'.`,
+        ],
     );
     return;
   }
 }
 
-function getNameAndLocation(node: babelTypes.Node): NamedIdentifier {
-  if (babelTypes.isClassMethod(node) || babelTypes.isClassProperty(node) ||
-      babelTypes.isObjectMethod(node) || babelTypes.isObjectProperty(node) ||
+function checkAllAdaptersAndFoundations(inputFilePath: string, inputCode: string) {
+  const isAdapterFile = inputFilePath.endsWith('adapter.ts');
+  const isFoundationFile = inputFilePath.endsWith('foundation.ts');
+  const publiclyExportedClassNames: string[] = [];
+  const ast = getAstFromCodeString(inputCode);
+
+  babelTraverse(ast, {
+    ClassDeclaration(nodePath: NodePath<babelTypes.ClassDeclaration>) {
```

Comment:
Don't we only ever have `interface` Adapter types in adapter modules? What are the other two traversals for? (Also, what's InterfaceDeclaration vs. TSInterfaceDeclaration?)

Reply:
Wait, I guess this happens not just for adapter files. I guess I find this confusing because both this and the below adapter-only traversal push into the same `publiclyExportedClassNames` array which seems error-prone in principle even if it's not right now.

(I don't suppose it'd be reasonable to have that function return an array rather than append to one...)